classes/class4/homework-soln/routes/index.js

```diff
@@ -0,0 +1,165 @@
+/* CONNECT TO MONGOOSE */
+var mongoose = require('mongoose');
+mongoose.connect(process.env.MONGOURI || 'mongodb://localhost/test');
+
+
+/* DEFINE SCHEMAS */
+var ingredientSchema = mongoose.Schema({
+	name: String,
+	price: Number,
+	inStock: Boolean
+});
+var Ingredient = mongoose.model('Ingredient', ingredientSchema);
+
+var orderSchema = mongoose.Schema({
+	customer: String,
+	ingredients: Array,
+	ingredientStr: String
+});
+var Order = mongoose.model('Order', orderSchema);
+
+
+/* DEFINE ROUTE CALLBACKS */
+var routes = {};
+
+routes.ingredients = function(req, res) {
+// query for in stock & out of stock ingredients separately and render
+	Ingredient.find({'inStock':true}, function(err, inData) {
+		Ingredient.find({'inStock':false}, function(err, outData) {
+			var hbsData = {'inStock':formatPrice(inData), 
+						   'outOfStock':formatPrice(outData)};
+			res.render('ingredients', hbsData);
+		});
+	});
+}
+
+routes.order = function(req, res) {
+// query for all ingredients and render
+	Ingredient.find(function(err, data) {
+		var hbsData = {'ingredients':formatPrice(data)};
+		res.render('order', hbsData);
+	});
+}
+
+routes.kitchen = function(req, res) {
+// query for all orders and render
+	Order.find(function(err, data) {
+		var hbsData = {'orders':data};
+		res.render('kitchen', hbsData);
+	});
+}
+
+routes.markOutOfStockPOST = function(req, res) {
```

Comment:
If you pulled this style of adding `POST` to the end of the function names from the class4 class-app, I just made it up there because I wanted to duplicated the functionality in both GET and POST. Otherwise this isn't really necessary.